sort_names.py

```diff
@@ -32,45 +34,90 @@ def get_short_list(filename, first_names, last_names):
-                and face_filter(row[30], limit_age, races)
-            ):
-                writer.writerow(row)
-                count += 1
+            if name_filter(row[24], first_names):
+                count_name -= 1
+                if row[25] not in last_names:
+                    count_last -= 1
+                    if position_filter(row[43], positions):
+                        count_pos -= 1
+                        if row[30] == "":
+                            count_no_link += 1
+                        else:
+                            if face_filter(row[30], limit_age, races):
+                                count_age_race -= 1
+                                count += 1
+                                writer.writerow(row)
             count_row += 1
             condition = round(100 * count_row / num_people)
 
-            if (condition % 5 == 0) and (n_cond != condition):
+            if (condition % 10 == 0) and (n_cond != condition):
                 n_cond = condition
                 print("---- {}% completed ----".format(n_cond))
         print("There were", num_people, "people before filtering")
         print("Done, now there are", count, "people")
         print("Results at the file '{}'".format(res_file_name))
 
+        print("Statistic:")
+        print("There are", count_no_link, "people without avatar")
+        print("Delete by Name filtered {} person".format(count_name))
+        print("Delete by Last name filtered {} person".format(count_last - count_name))
+        print("Delete by Position filtered {} person".format(count_pos - count_last))
+        print(
+            "Delete by DeepFace filtered {} person".format(
+                count_age_race - count_pos - count_no_link
+            )
+        )
+
+
+def download_img(imgURL):
+    urllib.request.urlretrieve(imgURL, "img.jpg")
+
 
 def input_black_list():
-    """Read .csv black list and return list of first_names and list of last_names"""
-    with open(black_list, "r", encoding="utf8") as file:
-        csvreader = csv.reader(file, delimiter=",")
-        first_names = []
-        last_names = []
-        title_t = next(csvreader)
-        for row in csvreader:
-            first_names.append(row[0])
-            last_names.append(row[1])
-    return first_names, last_names
-
-
-def face_filter(img, age=25, race="white"):
+    """Read Google Sheet black list and return list of first_names and list of last_names
+
+    black_list = "https://docs.google.com/spreadsheets/d/1lXjDSK5398p1yJuTJn3vO4Ux9KDJtqP1esar1Z9U4vc/edit#gid=0"
+    """
+    sheet_id = "1lXjDSK5398p1yJuTJn3vO4Ux9KDJtqP1esar1Z9U4vc"
+    df = pd.read_csv(
+        f"https://docs.google.com/spreadsheets/d/{sheet_id}/export?format=csv"
+    )
+    first_names = df[df.columns[0]].values.tolist()
+    last_names = df[df.columns[1]].values.tolist()
+    positions = df[df.columns[2]].values.tolist()
+    return first_names, last_names, positions
+
+
+def name_filter(name, black_list):
```

Comment:
please extract filters in the separate module, filters.py

Reply:
Ok, thank you!